api/server/router/container/container_routes.go

```diff
@@ -563,6 +563,11 @@ func (s *containerRouter) postContainersCreate(ctx context.Context, w http.Respo
 		hostConfig.ConsoleSize = [2]uint{0, 0}
 	}
 
+	if hostConfig != nil && versions.LessThan(version, "1.43") {
+		// Ignore OCIAnnotations because it was added in API v1.43.
+		hostConfig.OCIAnnotations = nil
+	}
```

Comment:
I'm following past precedent by ignoring the field on create for older API versions but unconditionally including it in the `/containers/{}/json` response for all API versions.